### DISCUSSION TITLE
Restrict access to apps based on LDAP groups

### DISCUSSION THREAD

**Discussion by radokristof** [nfrastack/container-lemonldap#42] [Q&A] [answered]
Dear All,

Does anyone know how I can restrict access to specific apps based on LDAP groups?
Mostly I use OIDC for integration, but this should be applied globally.

For example authentication is only allowed for an application if the user is in a specific OU in LDAP.

Thank you!

**Comment by tiredofit** [comment]
I've done it multiple ways throughout the years but I believe the recommended approach now is to use `inGroup`. https://lemonldap-ng.org/documentation/latest/rules_examples.html?highlight=ingroup

I've not done filters based on someones user entity residing in a specific OU, but made sure I added the user entity as a member of a specific group using the ldap memberOf overlay...

This works for Vhosts, SAML, and OIDC.

**Reply by radokristof** [reply]
Thank you! Sorry, I was a little bit generic, I'm indeed using memberOf overlay and in reality users are members of these groups.
However I have another problem with this inGroup approach. Is it somehow possible to limit the group search base based on the application?
Default, it will return all membership from LDAP, so for example I have `ou=admin,ou=app1,dc=example,dc=com` and `ou=admin,ou=app2,dc=example,dc=com`, it will return `admin` membership twice (it only returns the `cn` value). Is it somehow possible to limit this for each app, and for `app1` it will only search for membership in `ou=app1,dc=example,dc=com`?

**Reply by tiredofit** [reply, marked as answer]
You'd probably need a custom ldap query to support that, or architect your LDAP directory in a different manner. From what I can see your ou's are the applications themselves, where as if they were a group they would potentially be 

cn=appname,ou=applications,dc=example,dc=com with memberships from either other groups or uid entities.

**Reply by radokristof** [reply]
Yes, it was designed like this because some apps allow to match different roles based on retrieved groups from SSO. However if this cannot be done easily and not the standard way to go, I would let this idea and only use SSO to specify generic allow/deny for an application based on group(s).

**Reply by radokristof** [reply]
One more question: where is it possible to configure this per SAML service / OIDC RP?
I don't see an option for access rules in the RP setting, do I need to create a virtual host for this?

**Reply by tiredofit** [reply]
I believe in versions LLNG < 2.0 you had to first setup Vhost rules to ensure SAML rules worked, however you should see an access rules section that allows to restrict the OIDC or SAML connection wihch works with inGroup or other types of rules.

**Reply by radokristof** [reply]
Unfortunately, I can't see such option:
![image](https://github.com/tiredofit/docker-lemonldap/assets/36210568/1e2c567d-4141-46b5-92f1-2ad97e8cb6f3)

**Reply by tiredofit** [reply]
One more level to look - check inside Security under Options..
